### PR TITLE
Add an option to limit Executor concurrency

### PR DESCRIPTION
**Pull request by naweiss**
Closes issue #703

### CODE REVIEW

**Comment by naweiss** [comment]
Originally I thought about solving this problem using a simple slicing of the jobs array. But, this has the disadvantage of having to wait for a long job until running the next batch of jobs.
So, ultimately I used `PromisePool` which generate the next job immediately when the previous ended.

**Comment by firecow** [comment]
Good idea to use a concurrency library.

Add a test and we are good to go 👍

**Comment by naweiss** [comment]
I had hard time writing a test that jobs runs sequentially instead of parallel. I ended up using the `__snapshot__` method used in the padding tests. hope it is fine

**Comment by firecow** [comment]
Yeah, its fine.

Let us rename the option to --concurrency or --concurrent 👍

**Comment by naweiss** [comment]
Do you know how I solve the following linting warning?
```
Warning:   18:42  warning  '_' is defined but never used    @typescript-eslint/no-unused-vars
```

**Comment by naweiss** [comment]
I have to fix some minor thing before the merge

**Comment by naweiss** [comment]
There was something messed up with the author of the commits. It is fixed now

File: src/executor.ts [hunk 1]
```diff
@@ -2,19 +2,23 @@ import chalk from "chalk";
...
         let startCandidates = [];
 
         do {
-            const promises = [];
             startCandidates = Executor.getStartCandidates(jobs, stages, potentialStarters, argv.manual);
-            for (const s of startCandidates) {
-                promises.push(s.start());
+            if (startCandidates.length > 0) {
+                await PromisePool
+                    .withConcurrency(argv.concurrency ?? startCandidates.length)
```
Comment (hverlin) [inline src/executor.ts:16]: suggestion: use `os.cpus().length - 1` as the default value? Pass `-1` for unlimited concurrency?

Comment (firecow) [reply src/executor.ts:16]: If developers have problems with concurrency they can explicitly set it.

People have been running with unlimited up until now, no need to change that default.

**Comment by naweiss** [comment]
Anything else I need to fix?

**Review by firecow — APPROVED** [review]
Nope all looks good, merging

**Pull request merged**